plugin-java/che-plugin-java-ext-jdt/che-jdt-ext-machine/src/main/java/org/eclipse/che/jdt/rest/RefactoringService.java

```diff
@@ -131,14 +162,37 @@ public RefactoringPreview getRefactoringPreview(RefactoringSession refactoringSe
         return manager.getRefactoringPreview(refactoringSession.getSessionId());
     }
 
+    /**
+     * Change enabled/disabled state of the corresponding refactoring change.
+     * @param state the state of refactoring change
+     * @throws RefactoringException when there are no corresponding refactoring session or refactoring change
+     */
     @POST
     @Path("change/enabled")
     public void changeChangeEnabledState(ChangeEnabledState state) throws RefactoringException {
         manager.changeChangeEnabled(state);
     }
 
+    /**
+     * Get refactoring change preview. Preview contains new and old content of the file
+     * @param change the change to get preview
+     * @return refactoring change preview
+     * @throws RefactoringException
+     */
+    @POST
+    @Path("change/preview")
+    @Produces("application/json")
+    @Consumes("application/json")
+    public ChangePreview getChangePreview(RefactoringChange change) throws RefactoringException {
```

Comment:
Is this method change the state of refactoring change preview? if not may be method should be GET